### DISCUSSION TITLE
Custom No-Permission messages for portals

### DISCUSSION THREAD

**Discussion by xmorose** [Zrips/CMI#6796] [Feature Requests]
Having custom no-perm messages for portals would be really useful, especially for tutorials when having a tutorial at the end trying to say "Talk to x first" or "you need to do x before entering this portal". Would simply be useful.

**Comment by CloudeLecaw** [comment]
You know you can use soecialzed command to do this?

First - disable the portal target
Then enable run commands with no target setting in the Portal.

Now use some specialized commands like
```
- perm:cmi.portal.arandomnane@! msg! [playerName] &cIt looks like your not permitted yet to enter. Ask the wise ild man about..may he use how to open the portal.
- perm:cmi.portal.arandomnane! asConsole! cmi tppos -p:[playerName] 200 60 -200 world 90 90
```
You could as -s if u need that ;D

Well the cords of tppos should be coords of your Teleport target of you portal.

Now when player would have permission `cmi.portal.arandomnane` he would get ported to the cord - else he gets the meassage that you defined above in the code.

Note: dont forget to place the npc that may talknto the player can give him the permission on click or as quest reward ;)   (could be done with CMI-IC + citicenz, or  with a quest plugin)
